implementations/rust/ockam/ockam_app_lib/src/enroll/enroll_user.rs

```diff
@@ -61,10 +70,10 @@ impl AppState {
         Ok(())
     }
 
-    async fn enroll_with_token(&self) -> Result<()> {
+    async fn enroll_with_token(&self) -> Result<bool> {
         if self.is_enrolled().await.unwrap_or_default() {
             debug!("User is already enrolled");
-            return Ok(());
+            return Ok(false);
```

Comment:
I'd be tempted to use an enum there for a slightly better understanding:
```
enum EnrollmentStatus {
    AlreadyEnrolled,
    NewlyEnrolled,
}

Reply:
I was tempted as well, but it would have been a superset of a Result, like

```rust
enum EnrollmentResult<T> {
  AlreadyEnrolled,
  PendingValidation,
  Successful,
  Err(T)
}
  ```